### PR TITLE
API Add unit testing, Travis CI and some getter/setters for the 'raw' superglobal use

### LINKED ISSUE #17: Needs tests

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #17 and resolves #8

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - DB=MYSQL CORE_RELEASE=3.1
+
+matrix:
+  include:
+    - php: 5.6
+      env: DB=PGSQL CORE_RELEASE=3.5
+
+
+before_script:
+  - composer self-update
```
Comment (dhensby) [inline .travis.yml:17]: this should be `composer self-update || true` - previously we've had problems where composer was unavailable and so this line failed and tests wouldn't run.

composer comes pre-installed and so this update is just a nice-to-have and shouldn't block tests if it fails

Comment (tractorcow) [reply .travis.yml:17]: Apparently composer does a self-update out of the box. see the travis log.

image

Comment (dhensby) [reply .travis.yml:17]: version 1.0.0 :/

Comment (dhensby) [reply .travis.yml:17]: It then does it again - you're right - we don't need this in our `before_script` anymore...

Comment (tractorcow) [reply .travis.yml:17]: Yeah we are too paranoid. :)

**Pull request merged**